move generate_code resources into tools/autograd library

dagitses opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #75463

This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

Commit:
move generate_code resources into tools/autograd library

This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

albanD commented on tools/setup_helpers/generate_code.py:
```diff
@@ -51,8 +52,8 @@ def generate_code(ninja_global: Optional[str] = None,
     data_dir = os.path.join(runfiles_dir, 'pytorch') if runfiles_dir else ''
-    autograd_dir = os.path.join(data_dir, 'tools', 'autograd')
     tools_jit_templates = os.path.join(data_dir, 'tools', 'jit', 'templates')
+    autograd_dir = os.fspath(pathlib.Path(__file__).parent.parent / "autograd/")
```
Wait, is the division operator actually overloaded for Path? haha

Also, is the `/` at the end needed? That might cause issues on windows no?

dagitses replied:
It is overloaded. Cute huh!

Trailing / doesn't seem to do anything with `os.fspath`, at least not in Linux or MacOS.

Example:
```
$ python3 -c 'import os, pathlib; print(os.fspath(pathlib.Path("/dev/null/")))'
/dev/null
```

albanD replied:
My worry is with windows that uses `\` as separator!
If it does nothing, we should remove it to avoid any issue.

dagitses replied:
removing.

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]

albanD reviewed (approved):
SGTM!

Commit:
Update on "move generate_code resources into tools/autograd library"


This better matches how the corresponding Bazel library does it.

Differential Revision: [D35480501](https://our.internmc.facebook.com/intern/diff/D35480501/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D35480501/)!

[ghstack-poisoned]
